live-event-app/live-event-app/Theming/MessageInputTheme.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  MessageInputTheme.swift
+//
+//  PubNub Real-time Cloud-Hosted Push API and Push Notification Client Frameworks
+//  Copyright © 2022 PubNub Inc.
+//  https://www.pubnub.com/
+//  https://www.pubnub.com/terms
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+import PubNubChat
+import PubNubChatComponents
+
+final class CustomMessageInputBarComponent: MessageInputBarComponent {
+  override func setupComponent() {
+    super.setupComponent()
+    inputTextView?.layer.cornerRadius = 8
+  }
+}
+
+let messageInputComponentTheme = MessageInputComponentTheme(
```

Comment:
creating global const cuts off any error handling for unwrapping properties like `UIColor(named: "MessageInput.BackgroundColor")!`. Why not to embed it in class which needs this theme? Additionally now whole project sees this.

Reply:
I just wanted to provide as little code as possible, that's why there's no additional wrapper for this variable like class or struct. Moreover, declaring a theme object takes a lot of lines so this also increases the length of the file you want to move the variable to. I realize my solution isn't perfect, perhaps we haven't suggested to our customers the preferred approach to provide custom theme objects. Anyway, I can align with any solution that will be agreed upon by our team.